File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0;
        int right = s.size() - 1;

        while(left < right){

            while(left < right && !isValid(s[left])){
                left++;
            }

            while(left < right && !isValid(s[right])){
                right--;
            }

            if(tolower(s[left]) != tolower(s[right])){
                return false;
            }
            
            left++;
            right--;
        }
        return true;
        
    }

    bool isValid(char c){
        if(('0' <= c && c <= '9') || ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z' )){
            return true;
        } else{
            return false;
        }
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        res = []
        q = deque([root])
        
        while q:
            level = []
            q_len = len(q)
            for _ in range(q_len):
                cur = q.popleft()
                level.append(cur)
                if cur.left:
                    q.append(cur.left)
                if cur.right:
                    q.append(cur.right)
            lev_length = len(level)
            node_to_add = level[lev_length - 1]
            res.append(node_to_add.val)
        
        return res
            
        